### PR TITLE
Items show what enchantments they give

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
None

#### Purpose of change
Items with obvious abilities should display them

#### Describe the solution
Will type more just leaving this up as draft for now

#### Additional context
RM13
image

sentinel cloak
image

EXO armor
image

Smart Gun
image

### CODE REVIEW

**Comment by GuardianDll** [comment]
Making a mutation, that is granted when you use an item, just to show the enchantment effect feels like extremely bulky and ineffective solution imo

**Comment by bombasticSlacks** [comment]
That's probably fair, I can think on it a bit and see if I can come up with something simpler.

**Comment by GuardianDll** [comment]
Before i thought just to add a `description` field into `"type": "enchantment"`, which shows somewhere in the item/mutation/effect etc - same thing, still huge space for work, but much simpler

**Comment by bombasticSlacks** [comment]
> Before i thought just to add a description field into "type": "enchantment", which shows somewhere in the item/mutation/effect etc - same thing, still huge space for work, but much simpler

That's the plan, you can add descriptions or name + description. And I'm going to have enchantments with names show up in the effects list of the @ menu, hover for their description. I was definitely making it too complicated overall.

**Comment by GuardianDll** [comment]
Oh, while we are here, is it possible to make the same field for relic_procgen_data?

File: data/mods/Aftershock/items/armor/exosuit/exosuit_modules.json [hunk 1]
```diff
@@ -458,7 +446,17 @@
     "color": "light_gray",
     "armor": [ { "encumbrance": 5, "covers": [ "arm_l", "arm_r", "hand_l", "hand_r", "leg_l", "leg_r", "foot_l", "foot_r" ] } ],
     "flags": [ "CANT_WEAR", "EXO_MEDIUM_GADGET" ],
-    "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "mutations": [ "exo_climb_mutation" ] } ] }
+    "relic_data": {
+      "passive_effects": [
+        {
+          "name": { "str": "climbing assist" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/armor/exosuit/exosuit_modules.json:452]:
```suggestion
          "name": { "str": "Climbing Assist" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    "condition": "ALWAYS",
+    "name": { "str": "Cold Barrier" },
+    "description": "Reduces incoming cold damage by 25%.",
+    "values": [ { "value": "ARMOR_COLD", "multiply": -0.25 } ]
+  },
+  {
+    "type": "enchantment",
+    "id": "afs_motion_sensor_wield",
+    "has": "WIELD",
+    "description": "Actively detects nearby movement.",
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:15]:
```suggestion
    "name": { "str": "Active Motion Detector" },
    "description": "Actively detects nearby movement.",
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    "name": { "str": "Super Heated Blade" },
+    "description": "The energy radiating from this weapon cuts and burns with ease.",
+    "values": [ { "value": "ITEM_DAMAGE_CUT", "add": 18 }, { "value": "ITEM_DAMAGE_HEAT", "add": 35 } ]
+  },
+  {
+    "type": "enchantment",
+    "id": "afs_ench_exo_big_carry",
+    "has": "WORN",
+    "condition": "ACTIVE",
+    "name": { "str": "load support module" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:33]:
```suggestion
    "name": { "str": "Load Support Module" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    "values": [ { "value": "CARRY_WEIGHT", "add": 20000 } ],
+    "description": "Provides 20kg of additional storage."
+  },
+  {
+    "type": "enchantment",
+    "id": "afs_ench_exo_small_carry",
+    "has": "WORN",
+    "condition": "ACTIVE",
+    "name": { "str": "small load support module" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:42]:
```suggestion
    "name": { "str": "Small Load Support Module" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    "values": [ { "value": "CARRY_WEIGHT", "add": 10000 } ],
+    "description": "Provides 10kg of additional storage."
+  },
+  {
+    "type": "enchantment",
+    "id": "afs_ench_exo_recoil",
+    "has": "WORN",
+    "condition": "ACTIVE",
+    "name": { "str": "recoil mitigation system" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:51]:
```suggestion
    "name": { "str": "Recoil Mitigation System" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    "values": [ { "value": "WEAPON_DISPERSION", "add": -50 } ],
+    "description": "Reduces weapon dispersion."
+  },
+  {
+    "type": "enchantment",
+    "id": "afs_ench_exo_forcefield_bash_cut",
+    "has": "WORN",
+    "condition": "ACTIVE",
+    "name": { "str": "low energy forcefield" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:60]:
```suggestion
    "name": { "str": "Low Energy Forcefield" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    "values": [ { "value": "ARMOR_BASH", "add": -25 }, { "value": "ARMOR_CUT", "add": -25 } ],
+    "description": "Reduces incoming bash and cut damage by 25."
+  },
+  {
+    "type": "enchantment",
+    "id": "afs_ench_exo_forcefield_ballistic_pierce",
+    "has": "WORN",
+    "condition": "ACTIVE",
+    "name": { "str": "high energy forcefield" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:69]:
```suggestion
    "name": { "str": "High Energy Forcefield" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    "values": [ { "value": "ARMOR_BULLET", "add": -40 }, { "value": "ARMOR_STAB", "add": -40 } ],
+    "description": "Reduces incoming pierce and bullet damage by 40."
+  },
+  {
+    "type": "enchantment",
+    "id": "afs_ench_exo_AR_module",
+    "has": "WORN",
+    "condition": "ACTIVE",
+    "name": { "str": "education assist" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:78]:
```suggestion
    "name": { "str": "Education Assist" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    "values": [ { "value": "LEARNING_FOCUS", "add": 15 } ],
+    "description": "Assists with learning and study."
+  },
+  {
+    "id": "ench_climate_control_all_permanent_improved",
+    "name": { "str": "improved complete climate control" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:84]:
```suggestion
    "name": { "str": "Improved Complete Climate Control" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    "name": { "str": "extreme complete climate control" },
+    "description": "You are wearing equipment that helps handle harsh climates.",
+    "type": "enchantment",
+    "has": "WORN",
+    "condition": "ALWAYS",
+    "values": [ { "value": "CLIMATE_CONTROL_HEAT", "add": 150 }, { "value": "CLIMATE_CONTROL_CHILL", "add": 150 } ]
+  },
+  {
+    "id": "ench_climate_control_all_permanent_space",
+    "name": { "str": "space climate control" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:102]:
```suggestion
    "name": { "str": "Space Climate Control" },
```

File: data/mods/Aftershock/items/item_enchants.json [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    "description": "You are wearing equipment that helps handle harsh climates.",
+    "type": "enchantment",
+    "has": "WORN",
+    "condition": "ALWAYS",
+    "values": [ { "value": "CLIMATE_CONTROL_HEAT", "add": 300 }, { "value": "CLIMATE_CONTROL_CHILL", "add": 300 } ]
+  },
+  {
+    "id": "ench_smart_gun",
+    "name": { "str": "smart gun" },
```
Comment (bombasticSlacks) [inline data/mods/Aftershock/items/item_enchants.json:111]:
```suggestion
    "name": { "str": "Smart Gun" },
```

**Comment by GuardianDll** [comment]
Btw, does this works only when enchantment is separated and has an id, or it can be used right inside the relic_data?

**Comment by bombasticSlacks** [comment]
yup there are examples in this PR

**Pull request merged**